mne/io/fiff/raw.py

```diff
@@ -117,19 +117,24 @@ def __init__(self, fname, allow_maxshield=False, preload=False,
                 n_samples += r.last_samp - r.first_samp + 1
 
         # Add annotations for in-data skips
-        offsets = [0] + self._raw_lengths[:-1]
-        for extra, first_samp, offset in zip(self._raw_extras,
-                                             self._first_samps, offsets):
-            for skip in extra:
-                if skip['ent'] is None:  # these are skips
-                    if self.annotations is None:
-                        self.set_annotations(Annotations((), (), ()))
-                    start = skip['first'] - first_samp + offset
-                    stop = skip['last'] - first_samp + offset
-                    self.annotations.append(
-                        _sync_onset(self, start / self.info['sfreq']),
-                        (stop - start + 1) / self.info['sfreq'],
-                        'BAD_ACQ_SKIP')
+        for extra in self._raw_extras:
+            start = np.array([_['first'] for _ in extra if _['ent'] is None])
+            stop = np.array([_['last'] for _ in extra if _['ent'] is None])
+            duration = (stop - start + 1) / self.info['sfreq']
+
+            annot = Annotations(onset=(start / self.info['sfreq']),
+                                duration=duration,
```

Comment:
as duration is not 0 and data are stuck together I am wondering if this is doing the right thing. Imagine there is an event just after the 'BAD_ACQ_SKIP' will it be dropped if duration is not zero. @larsoner thoughts?

Reply:
Agreed that if you concat two raws, each of length 1000 samps, you should be able to get an `epochs` object with the first 1000 samps and the second 1000 samps.

I thought we had a test like this already in `master` but maybe not. If we don't, we should:

1. make a PR to add it and make it work properly (if it does not exist or work properly already)
2. rebase this PR

Reply:
agreed.

Reply:
This code replicates what was in the code base:
```py
                    self.annotations.append(		￼
￼                        _sync_onset(self, start / self.info['sfreq']),
￼                        (stop - start + 1) / self.info['sfreq'],		￼                
￼                        'BAD_ACQ_SKIP')
```

We had that conversation in gitter:

@massich
> I'm trying to understand this part of the code base:
https://github.com/mne-tools/mne-python/blob/6aff6e6cc11e7428fe6a89bf8516d9b3669d3938/mne/io/fiff/raw.py#L119-L132
>
> let's say that we have an experiment with several parts (XXX recording with A
annotations, and YYY with B) that gets recorded in `foo_raw.fif` and `bar_raw.fif`,
and we do something like `raw = read_raw_fif([foo_raw.fif, bar_raw.fif])`.
If I understand the snipped properly, the raw file has X and Y stick together
in `_data`, B onset is modified and adds an annotation C where X and Y are stick
which lasts a duration (time between X and Y).
>
> this picture my help
```
              meas-date
experiment        |---XXXXXXXXXXXXXXXX------YYYYYYYYYYY-----
                  |------AAA-------------------BB-----------

foo_raw.fif       |---XXXXXXXXXXXXXXXX----------------------
                  |------AAA--------------------------------

bar_raw.fif       |-------------------------YYYYYYYYYYY-----
                  |----------------------------BB-----------

raw-object        |---XXXXXXXXXXXXXXXXYYYYYYYYYYY----
                  |------AAA-------------------------
                  |----------------------BB----------
                  |-------------------CCCCCC---------
```

@larsoner:
> `_raw_extra` holds whatever file-format-specific information is necessary to properly read data
> your example is wrong
> it should be
```
raw-object        |---XXXXXXXXXXXXXXXX------YYYYYYYYYYY----
                  |------AAA-------------------------------
                  |----------------------------BB----------
                  |-------------------CCCCCC---------------
```
> but this doesn't usually happen with `concatenate_raws`
> some formats like `neuromag` allow you to turn recordings off *in the middle of a file* then turn them back on
> so with no annotations added at all you can have files that are like:
```
---XXXXXXXXX----XXXXXXXXXX
------------SSSS----------
```

Reply:
FYI @agramfort @massich there is already a test for the zero-sample `BAD_ACQ_SKIP` that you get here:

https://github.com/mne-tools/mne-python/blob/master/mne/tests/test_annotations.py#L368-L379

I'll push a tiny commit to this PR to enhance the test.

Reply:
... correction, it's not BAD_ACQ_SKIP but a special `BAD boundary` event with zero duration